oe_webtools_analytics/src/EventSubscriber/AnalyticsEventSubscriber.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types = 1);
+
+/**
+ * @file
+ * Listening to the AnalyticsEvent.
+ */
+
+namespace Drupal\oe_webtools_analytics\EventSubscriber;
+
+use Drupal\oe_webtools_analytics\AnalyticsEventInterface;
+use Drupal\Core\Url;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Drupal\oe_webtools_analytics\Event\AnalyticsEvent;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * Event Subscriber AnalyticsEventSubscriber.
+ */
+class AnalyticsEventSubscriber implements EventSubscriberInterface {
+  /**
+   * The Configuration overrides instance.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $config;
+
+  /**
+   * {@inheritdoc}
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $requestStack;
+
+  /**
+   * AnalyticsEventSubscriber constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   The configuration object.
+   * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
+   *   The request on the stack.
+   */
+  public function __construct(ConfigFactoryInterface $configFactory, RequestStack $requestStack) {
+    // Get id from settings.php!
```

Comment:
Instead of using if/else, wouldn't it be better to, first check for the availability of the settings, then react accordingly ?
This would reduce overall code complexity.

```php
if (settings_not_available) {
  throw Exception();
}

// rest of the code here.
```